src/msrc_file_system_depth_device.cpp

```diff
@@ -0,0 +1,69 @@
+#include <boost/filesystem.hpp>
+#include <opencv2/opencv.hpp>
+#include <seng499/msrc_file_system_depth_device.hpp>
+#include <regex>
+#include <iostream>
+#include <cstdint>
+
+namespace seng499 {
+
+	static const std::regex & get_regex() {
+
+		static const std::regex msrc_depth_file_pattern(".*depth.*");
+		return msrc_depth_file_pattern;
+
+	}
+	
+	
+	bool msrc_file_system_depth_device_filter::operator () (const boost::filesystem::path & path) const {
+		
+		return std::regex_match(path.filename().string(), get_regex());
+		
+	}
+	
+	
+	std::vector<float> msrc_file_system_depth_device_frame_factory::operator () (const boost::filesystem::path & path, std::vector<float> vec) {
+	
+		// make sure this is empty
+		vec.clear();
+
+		// reserve the size of our image
+		vec.reserve(640*480);
+
+		// Load a grayscale depth image. Anydepth is necessary as these are 16 bit ints 
+		cv::Mat img = cv::imread(path.native(), CV_LOAD_IMAGE_GRAYSCALE | CV_LOAD_IMAGE_ANYDEPTH);
+
+		// Convert mm to m, cast uint16 to float and push into the vector		
+		for (int i=0; i < img.rows; i++)
+			for (int j = 0; j < img.cols; j++)
+				vec.push_back(float(img.at<uint16_t>(i,j)/1000.0f));
```

Comment:
Should be `std::uint16_t`.

The standard requires that `<cstdint>` pull in everything from `<stdint.h>` in the `std` namespace.  It allows `<cstdint>` to pull in the un-namespaced versions but doesn't require it, so it's best not to rely on it.